sample-app/src/components/SearchBar.tsx

```diff
@@ -6,28 +6,50 @@ import '../sass/SearchBar.scss';
 import '../sass/Autocomplete.scss';
 import LoadingIndicator from './LoadingIndicator';
 import { useAutocomplete } from '../hooks/useAutocomplete';
+import SearchHandler from '../utils/searchhandler';
+import { SearchIntent } from '@yext/answers-headless';
 
 const SCREENREADER_INSTRUCTIONS = 'When autocomplete results are available, use up and down arrows to review and enter to select.'
 
 interface Props {
   placeholder?: string,
   isVertical: boolean,
+  geolocationOptions?: PositionOptions,
   screenReaderInstructionsId: string
 }
 
 /**
  * Renders a SearchBar that is hooked up with an Autocomplete component
  */
-export default function SearchBar({ placeholder, isVertical, screenReaderInstructionsId }: Props) {
+export default function SearchBar({
+  placeholder,
+  isVertical,
+  geolocationOptions,
+  screenReaderInstructionsId 
+}: Props) {
   const answersActions = useAnswersActions();
   const query = useAnswersState(state => state.query.input);
-  const [ autocompleteResults, executeAutocomplete ] = useAutocomplete(isVertical);
   const isLoading = useAnswersState(state => state.searchStatus.isLoading);
+  const [ 
+    autocompleteResponse,
+    responseToLatestRequestRef,
+    executeAutocomplete
+  ] = useAutocomplete(isVertical);
 
-  function executeQuery () {
-    isVertical
-      ? answersActions.executeVerticalQuery()
-      : answersActions.executeUniversalQuery();
+  async function executeQuery () {
+    const responseToLatestRequest = await responseToLatestRequestRef.current;
+    if (responseToLatestRequest?.inputIntents.includes(SearchIntent.NearMe)) {
```

Comment:
can we extract this method (except for waiting on the autocomplete request on line 40) somewhere, since it's being used in a number of places? I was thinking `executeSearchWIthIntents(intents: SearchIntent[])` and we could move it to `utils/executeSearchWithIntents.ts`?

Reply:
sure thing, added to SearchHandler for now, but if we decide to go with separate functions than I will move it out to another file